go/arrow/ipc/writer.go

```diff
@@ -709,6 +709,21 @@ func (w *recordEncoder) visit(p *Payload, arr arrow.Array) error {
 		}
 		w.depth++
 
+	case *arrow.RunEndEncodedType:
+		arr := arr.(*array.RunEndEncoded)
+		w.depth--
+		child := arr.LogicalRunEndsArray(w.mem)
```

Comment:
So this can create a new array if the offset is non-zero?

Reply:
yea. if the offset is 0 then it's a slice, if it's non-zero it's a new array because the actual run-ends need to get adjusted to provide an offset of 0